hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/ConfigProvider.java

```diff
@@ -102,4 +105,35 @@ public static ClientConfig locateAndGetClientConfig() {
         return config;
     }
 
+
+    @Nonnull
+    public static Config locateAndGetMemberConfig(@Nullable Properties properties) {
+        Config config;
+        XmlJetMemberConfigLocator xmlConfigLocator = new XmlJetMemberConfigLocator();
+        YamlJetMemberConfigLocator yamlConfigLocator = new YamlJetMemberConfigLocator();
+
+        if (yamlConfigLocator.locateFromSystemProperty()) {
+            // 1. Try loading config if provided in system property and it is an YAML file
+            config = new YamlConfigBuilder(yamlConfigLocator.getIn()).setProperties(properties).build();
```

Comment:
does yaml have priority over xml ?

Reply:
This is related to https://github.com/hazelcast/hazelcast-jet/issues/1380, if it doesn't have yaml extensions we are treating it as XML one. So we are checking if the file has yaml extensions first.